packages/common/src/bls/ICalldataCacheDecoderInteractor.ts

```diff
@@ -2,7 +2,10 @@ import { PrefixedHexString } from 'ethereumjs-util'
 import BN from 'bn.js'
 
 export interface AddressesCachingResult {
-  ids: BN[][]
+  senderAsIds: BN[],
```

Comment:
It should be separated to Gateway Address Caching Result and ERC20 Address Caching Result as it is a different object for different caching purposes